src/jormungandr/testing/jormungandr-integration-tests/src/networking/p2p/stats.rs

```diff
@@ -159,6 +163,66 @@ macro_rules! build_network {
     }};
 }
 
+#[test]
+#[ignore]
+pub fn test_gateway_node_not_publishing() {
+    let mut network_controller = build_network!()
+        .topology(
+            Topology::default()
+                .with_node(Node::new(LEADER))
+                .with_node(Node::new(LEADER_CLIENT).with_trusted_peer(LEADER)),
+        )
+        .wallet_template(
+            WalletTemplateBuilder::new("alice")
+                .with(1_000_000)
+                .delegated_to(LEADER)
+                .build(),
+        )
+        .wallet_template(WalletTemplateBuilder::new("bob").with(1_000_000).build())
+        .blockchain_config(BlockchainConfiguration::default().with_leader(LEADER))
+        .build()
+        .unwrap();
+
+    let leader = network_controller
+        .spawn(SpawnParams::new(LEADER).in_memory())
+        .unwrap();
+
+    //
+    // spin up gateway node
+    //
+    let mut gateway_node_binary = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+
+    gateway_node_binary.pop();
+    gateway_node_binary.pop();
+    gateway_node_binary.pop();
+    gateway_node_binary.pop();
+    gateway_node_binary.push("target/debug/gateway");
+
+    let leader_client = network_controller
+        .spawn(SpawnParams::new(LEADER_CLIENT).jormungandr(gateway_node_binary))
+        .unwrap();
+
+    let mut alice = network_controller.controlled_wallet("alice").unwrap();
+    let mut bob = network_controller.controlled_wallet("bob").unwrap();
+
+    let fragment_sender = FragmentSender::from(&leader.rest().settings().unwrap());
+
+    // gateway node should not be able to send fragments
+    match fragment_sender.send_transactions_round_trip(
+        5,
+        &mut alice,
+        &mut bob,
+        &leader_client,
+        100.into(),
+    ) {
+        Ok(_) => panic!("gateway node should not be able to send fragments"),
+        Err(err) => assert_eq!(
+            err.to_string(),
+            "Too many attempts failed (1) while trying to send fragment to node: ".to_string()
+        ),
+    };
+}
+
```

Comment:
i would add one more test just to be sure that gateway node function correctly with trusted nodes. For example we are configure gateway node not to publish but we need to make sure that it still sync with rest of the network correctly